source/server/backtrace.h

```diff
@@ -83,6 +83,23 @@ class BackwardsTrace : Logger::Loggable<Logger::Id::backtrace> {
   }
 
   /**
+   * Capture a stack trace from a particular address.
+   *
+   * This can be used to capture a useful stack trace from a fatal signal
+   * handler.
+   *
+   * @param address The stack trace will begin from this address.
+   */
+  void CaptureFrom(void* address) {
+#ifdef NDEBUG
```

Comment:
Is it OK if backtraces are always logged at critical level? The point of the log_in_prod uglyness was to work around the fact that we don't actually emit logs below that level in optimized builds, and this feature should still be useful for investigating issues in optimized builds.  In practice with the signal handler either something has gone very wrong, or, a developer has manually added a statement to assist in debugging.  In both cases critical seems fine to me.